server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -285,6 +285,18 @@ public class IndicesService extends AbstractLifecycleComponent
         Property.Dynamic
     );
 
+    /**
+     * This setting is used to restrict creation or updation of index where the `index.translog.durability` index setting
+     * is set as ASYNC if enabled. If disabled, any of the durability mode can be used and switched at any later time from
+     * one to another.
+     */
+    public static final Setting<Boolean> CLUSTER_REMOTE_INDEX_RESTRICT_ASYNC_DURABILITY_SETTING = Setting.boolSetting(
+        "cluster.remote_store.index.restrict.async-durability",
```

Comment:
should we do something like `"cluster.remote_store.index.restrict.durability": "async"` ,  instead of a boolean. 

This would allow us to restrict sync durability as well in future if required.